internal/signaling/handler.go

```diff
@@ -120,9 +120,16 @@ func Handler(ctx context.Context, store stores.Store, cloudflare *cloudflare.Cre
 				if err != nil {
 					util.ReplyError(ctx, conn, err)
 				} else {
+					requestPacket := CredentialsPacket{}
+					if err := json.Unmarshal(raw, &requestPacket); err != nil {
+						util.ReplyError(ctx, conn, err)
+						continue
+					}
+
```

Comment:
Instead of unmarshaling again, we can also include the rid in the `typeOnly` struct above on line 107
Then that would become:
```go
typeOnly := struct{
  Type string      `json:"type"`, 
  RequestID string `json:"rid"`,
}{}
```
And then we might want to rename it to `base` instead of typeOnly.